ospfd/ospf_vty.c

```diff
@@ -2058,6 +2058,17 @@ DEFUN (ospf_abr_type,
 	if (ospf->abr_type != abr_type) {
 		ospf->abr_type = abr_type;
 		ospf_schedule_abr_task(ospf);
+
+		/* Changing to standard or shortcut. The ABR task might not
+		 * initiate SPF recalculation if OSPF flags remain the same.
+		 * And inter-area routes would not be added/deleted according
+		 * to the new ABR type. So this needs to be done here.
+		 */
+		if (abr_type == OSPF_ABR_STAND
```

Comment:
Any change to the ABR type would impact the route calculation? Why do you think otherwise?

Reply:
My tests show this. When changing between standard and shortcut , the router remains ABR, the OSPF flags don't change, and SPF recalculation doesn't occur (routes from summary LSAs are not processed). Please see `ospf_check_abr_status()`. At the same time, when changing from standard/shortcut to cisco/ibm (or vice versa), the router stops being ABR, the OSPF flags change, and SPF recalculation occurs.

Reply:
Look at ospf_ia_routing() in ospf_ia.c as a reference.

Reply:
I'm sorry I don't quite understand what changes you are proposing. If you are proposing to remove the ABR check and run SPF recalculation unconditionally, I think this would be extra work. My tests show that the problematic case is change between standard and shortcut because of lack of SPF recalculation. Other cases already include SPF recalculation. So running SPF recalculation unconditionally for them would cause the procedure to happen twice: here in the command handler and in `ospf_check_abr_status()`.

If that isn't what changes you are proposing, please clarify your proposal.

Reply:
Based on the current design,  I don't think  it is going to trigger SPF if the ABR type changed from cisco/IBM to STAND/shortcut . Because, here SPF is triggering only when ABR flag changed (set<=>unset). But changing  ABR type will not impact if it is already an ABR.

Reply:
That is exactly what I'm proposing - an SPF calculation is needed anytime the area type changes. If you have the right topology with multiple areas, any of these changes could result in an different SPF outcome. Since the intra-area SPF is different for shortcut areas a change to/from OSPF_ABR_SHORTCUT will result in difference routes installed. So for intra-area SPF, you need to check if the previous or new abr_type is/was OSPF_ABR_SHORTCUT. For all the other transitions, there will be different logic for the calculation and it MAY result in different routes - look at ospf_ia_routing() in ospf_ia.c and tell me that changing the abr_type doesn't change the inter-area route calculation.